### PR TITLE
Switch to openshift/api

### PR DESCRIPTION
**Pull request by deads2k**
Starts vendoring github.com/openshift/api instead of using the native types.  I haven't updated unit tests yet, but I could use some CI runs to chase problems.

### CODE REVIEW

**Comment by deads2k** [comment]
@liggitt This is actually pretty close.  I'm chasing a failure in test/cmd/image, but that's the last one I know of.

**Comment by deads2k** [comment]
Oh, and we'll fix the proto definition stuff later.  No idea why it dropped out.

**Comment by deads2k** [comment]
I'm missing https://github.com/openshift/origin/pull/17091/files

**Comment by deads2k** [comment]
@mfojtik I'm likely to get this ready tomorrow am.  ptal

**Comment by mfojtik** [comment]
@deads2k i'm going to try to chase the cmd failure till you get in.

File: pkg/image/apis/image/docker10/doc.go [hunk 1]
```diff
@@ -1,4 +1,2 @@
-// +k8s:deepcopy-gen=package,register
```
Comment (mfojtik) [inline pkg/image/apis/image/docker10/doc.go]: why you dropped this?

File: pkg/network/apis/network/v1/doc.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
-// +k8s:deepcopy-gen=package,register
 // +k8s:conversion-gen=github.com/openshift/origin/pkg/network/apis/network
+// +k8s:conversion-gen-external-types=github.com/openshift/api/network/v1
 // +k8s:defaulter-gen=TypeMeta
-// +k8s:openapi-gen=true
+// +k8s:defaulter-gen-input=../../../../../../../../github.com/openshift/api/network/v1
```
Comment (mfojtik) [inline pkg/network/apis/network/v1/doc.go]: this is pretty ugly... can we improve the generator?

**Comment by mfojtik** [comment]
Can't repro this locally with this PR rebased against master, so I guess you just need a rebase and we are good to go.

/retest

**Comment by mfojtik** [comment]
@deads2k seems like `hack/verify-generated-swagger-descriptions.sh` is gone

File: pkg/network/apis/network/v1/doc.go (hunk 2 above)
Comment (deads2k) [reply pkg/network/apis/network/v1/doc.go]: > this is pretty ugly... can we improve the generator?

Yeah.  This one in particular is off.  Once we break the back of the problem we can fine tune.

File: pkg/image/apis/image/docker10/doc.go (hunk 1 above)
Comment (deads2k) [reply pkg/image/apis/image/docker10/doc.go]: > why you dropped this?

it is generated in openshift/api.

**Comment by deads2k** [comment]
> @deads2k seems like hack/verify-generated-swagger-descriptions.sh is gone

Yeah.  Also done in openshift/api

**Comment by deads2k** [comment]
Got an lgtm via irc.  tagging and bumping priority.

**Comment by liggitt** [comment]
all blocking CI jobs were green on the PR HEAD commit
PR is rebased on current `master`
manually merging to unblock other PRs this would conflict with

**Pull request merged**